src/security/ldap/src/main/java/org/geoserver/security/ldap/LDAPRoleService.java

```diff
@@ -287,7 +295,7 @@ public GeoServerRole getParentRole(GeoServerRole role) throws IOException {
 
     @Override
     public GeoServerRole getRoleByName(String role) throws IOException {
-        if (role.startsWith("ROLE_")) {
+        if (role.startsWith(rolePrefix)) {
             // remove standard role prefix
             role = role.substring(5);
```

Comment:
If the role prefix is configured, it might not be 5 chars long. Use its actual lenght.

Reply:
fixed